### PR TITLE
[do not merge] Add workstation and spark compute to MachineLearningServices

### PR DESCRIPTION
**Pull request by paulshealy1**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anuchandy** [comment]
@paulshealy1 could you take a look at the SDK errors (1 new SDK Error) and warnings (12 new SDK warnings) , reported here https://github.com/Azure/azure-rest-api-specs/pull/5526/checks?check_run_id=90516647?

**Comment by dsgouda** [comment]
startbuild

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 1]
```diff
@@ -833,6 +939,329 @@
...
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "The operation was successful.",
+              "schema": {
+                "$ref": "#/definitions/Workstation"
+              }
+            },
+            "201": {
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Is a 202 a better option?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 2]
```diff
@@ -1620,6 +2077,166 @@
...
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "description": "AML Spark Compute properties",
+              "type": "object",
+              "properties": {
+                "sparkVersion": {
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Is this finite? Should it be an enum or is it open?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 3]
```diff
@@ -1620,6 +2077,166 @@
...
+                },
+                "userAccountCredentials": {
+                  "title": "User account credentials.",
+                  "description": "Credentials for an administrator user account that will be created on each compute node.",
+                  "$ref": "#/definitions/UserAccountCredentials"
+                },
+                "subnet": {
+                    "title": "Subnet.",
+                    "description": "Virtual network subnet resource ID the compute nodes belong to.",
+                    "$ref": "#/definitions/ResourceId"
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: I assume a linked access check will be defined for this?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 4]
```diff
@@ -2177,6 +2966,264 @@
...
+    "PrivateRegistryCredentials": {
+      "properties": {
+        "username": {
+          "type": "string",
+          "title": "User name.",
+          "description": "User name to login to the repository."
+        },
+        "password": {
+          "type": "string",
+          "title": "Password.",
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Secret is not returned on GET, right?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 5]
```diff
@@ -2177,6 +2966,264 @@
...
+          "description": "The continuation token."
+        }
+      },
+      "description": "Values returned by the workstation list operation."
+    },
+    "NetworkProfile": {
+      "description": "Network profile for the workstation",
+      "properties": {
+        "subnetId": {
+          "description": "Subnet ID",
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: I assume a linked access check will be defined for this?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 6]
```diff
@@ -2177,6 +2966,264 @@
...
+          "$ref": "#/definitions/ResourceId"
+        },
+        "networkInterface": {
+          "description": "The network interface",
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: I assume a linked access check will be defined for this?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Some minor comments to review.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 2 above)
Comment (pradeepgunda) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: No, this list keeps growing and we don't want the swagger/SDK update on new spark version support.. That's why we modelled as string.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 3 above)
Comment (pradeepgunda) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: yes. we have linked access check for this in our manifest file

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 4 above)
Comment (pradeepgunda) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Yes, secret is not returned on GET.

**Comment by anuchandy** [comment]
Requesting reviewer reassign since I'm OOF

**Comment by salameer** [comment]
Hi @jhendrixMSFT , reasssigining this to you since Anu is out

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 7]
```diff
@@ -0,0 +1,3261 @@
...
+        },
+        "createdOn": {
+          "type": "string",
+          "description": "The date and time when the compute was created.",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "modifiedOn": {
+          "type": "string",
+          "description":
```
Comment (jhendrixMSFT) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: This is missing the `computeType` discriminator property and is the reason the builds are failing.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 8]
```diff
@@ -0,0 +1,3261 @@
...
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/UnderlyingResourceActionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (jhendrixMSFT) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Is there a 204 for the non-existent case (similar to `Workspaces_Delete`)?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Discriminator must be fixed.

**Comment by jhendrixMSFT** [comment]
@paulshealy1 hello any update?

**Comment by paulshealy1** [comment]
> @paulshealy1 hello any update?

We are finalizing a couple of changes on our side, and we will get back to you with those and resolutions to the comments.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 5 above)
Comment (paulshealy1) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Yes, we will add a linked access check for this in our manifest file.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 6 above)
Comment (paulshealy1) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Yes, we will add a linked access check for this in our manifest file.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 3 above)
Comment (ravbhatnagar) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: common pattern is to use Id suffix for such property names, so subnetId is preferred.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 6 above)
Comment (ravbhatnagar) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: common pattern is to use Id suffix for such property names, so networkInterfaceId is preferred.

**Comment by ravbhatnagar** [comment]
Looks good from ARM side. Signing off.

**Pull request closed without merging**